app/main/views/frameworks.py

```diff
@@ -207,3 +219,20 @@ def _framework_updates_page(error_message=None):
         error_message=error_message,
         **template_data
     ), status_code
+
+
+def _get_declaration_status():
```

Comment:
I think it's a good idea to limit `views.py` to view functions, especially once the file grows large. It makes it a bit easier to audit routes/decorators etc.

API calls from helpers shouldn't be a problem as long as you pass in the `apiclient` as an argument: that way we don't need to mock any additional references and it's a bit more explicit. I think we've done this in a few other places.